BrainCloudClient/Assets/BrainCloud/Client/BrainCloud/BrainCloudWrapper.cs

```diff
@@ -441,6 +445,27 @@ public void InitWithApps(string url, string defaultAppId, Dictionary<string, str
         LoadData();
     }
 
+    
+    /// <summary>
+    /// Resets the wrapper.
+    /// Since the WrapperName is set upon re-initialization of the wrapper, the name is reset by choice here. As the user
+    /// may want to reset the wrapper's fields without also restting the name. 
+    /// </summary>
+    public void resetWrapper(bool resetWrapperName = false)
+    {
+        ResetStoredAnonymousId();
+        ResetStoredAuthenticationType();
+        ResetStoredProfileId();
+        _wrapperData = new WrapperData();
+        Client.Wrapper = null;
```

Comment:
No ResetCommunications? need to kill sockets and such if relay/RTT are alive.

Reply:
the init call for the wrapper automatically passes through the init call for the client, and in the client init we call resetCommunication, so it passes through the reset of rtt and relay

Reply:
Ok, but here we're in a call called `resetWrapper`. It should reset everything. Not depend on "oh this is called before maybe" :)

Reply:
thats fair